### PR TITLE
[release-4.7] Bug 1941574: test/extended/router/idle: skip test on virt platforms

### PR DESCRIPTION
**Pull request by frobware**
This is a manual backport of PR #25924.

### CODE REVIEW

**Comment by Miciah** [comment]
> This is a manual backport of PR #25924.

Looks like it bundles https://github.com/openshift/origin/pull/25874; is that intentional?

**Comment by Miciah** [comment]
/lgtm
although I would like to sort out my question in https://github.com/openshift/origin/pull/25999#issuecomment-804703106, and I cannot approve changes in this branch.

**Comment by frobware** [comment]
> > This is a manual backport of PR #25924.
> 
> Looks like it bundles #25874; is that intentional?

No, that wasn't intentional. Thanks for catching this.

I was debugging a related 4.7 backport to this test (https://github.com/openshift/origin/pull/25892) which is currently failing in CI and trying to work out whether I needed both for the idle tests to pass 4.7 CI.

Updated.

File: test/extended/router/idle.go [hunk 1]
```diff
@@ -37,14 +38,24 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 
 	g.Describe("The HAProxy router", func() {
 		g.It("should be able to connect to a service that is idled because a GET on the route will unidle it", func() {
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to get cluster-wide infrastructure")
+			switch infra.Status.PlatformStatus.Type {
+			case configv1.OvirtPlatformType, configv1.KubevirtPlatformType, configv1.LibvirtPlatformType, configv1.VSpherePlatformType:
```
Comment (candita) [inline test/extended/router/idle.go:44]: How about None, OpenStack, and *metal platformtypes?

Comment (candita) [reply test/extended/router/idle.go:44]: As discussed, this can be considered in 4.8.

**Comment by frobware** [comment]
@Miciah @knobunc fixed merge conflicts.  Please could we label and approve again. Thanks.

**Pull request merged**